tools/CompileReleaseNotes.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import collections
+import git
+import logging
+import os
+import re
+import textwrap
+from typing import List
+
+logger = logging.getLogger(__name__)
+
+
+def get_last_release(repo: git.Repo, head_rev: str) -> git.TagReference:
+    """Retrieve the release closest to the `head_rev` in the git history
+
+    Returns:
+      The tag representing the latest release, as measured by number of commits
+      from the `head_rev`.
+    """
+    def is_version_tag(tag):
+        return str(tag).startswith('v')
+
+    def distance_from_head(tag):
+        return len(list(repo.iter_commits(rev='{}..{}'.format(tag, head_rev))))
+
+    return sorted(filter(is_version_tag, repo.tags), key=distance_from_head)[0]
+
+
+PullRequest = collections.namedtuple('PullRequest', ['id', 'title'])
+
+
+def get_merged_pull_requests(repo: git.Repo, from_rev: str,
+                             to_rev: str) -> List[PullRequest]:
+    """Parses list of merged PRs from merge commits.
+
+    Parses merge commits in the repository between the revisions `from_rev` and
+    `to_rev`. This is faster than querying GitHub and we can filter by commit
+    SHAs instead of date.
+
+    Returns:
+      Merged pull-requests, ordered from most recently merged to least recently
+      merged.
+    """
+    merge_commit_msg_pattern = '^Merge pull request #([0-9]+) from'
+    merged_prs = []
+    for commit in repo.iter_commits(rev='{}..{}'.format(from_rev, to_rev)):
+        merge_commit_match = re.match(merge_commit_msg_pattern,
+                                      commit.message,
+                                      flags=re.MULTILINE)
+        if not merge_commit_match:
+            continue
+        merged_prs.append(
+            PullRequest(id=int(merge_commit_match.group(1)),
+                        title=' '.join(commit.message.splitlines(False)[2:])))
+    return merged_prs
+
+
+if __name__ == "__main__":
+    # The release notes always refer to the repository that contains this file
+    repo = git.Repo(os.path.dirname(os.path.dirname(__file__)))
```

Comment:
Does the python interface provide something equivalent to `git rev-parse --show-toplevel`?

Reply:
Yes! Fixed.